packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1663,11 +1664,8 @@ export function flushSyncWork(): boolean {
 
 export function isAlreadyRendering(): boolean {
   // Used by the renderer to print a warning if certain APIs are called from
-  // the wrong context.
-  return (
-    __DEV__ &&
```

Comment:
is removing the `__DEV__` check intentional? looks safe since all existing callsites have DEV checks

Reply:
Maybe instead it should add a `__PROFILE__` check with `__DEV__`?

Reply:
It's intentional. There's nothing about this check that's really DEV only and we actually have this same check in a lot of prod cases within this same file which could really just use this same helper.

The reason I needed to relax it now is that I needed in it in a different file.